### PR TITLE
Duplicate errors that are un-writable

### PR DESCRIPTION
**Pull request by jridgewell**
Certain errors (`DOMException`s, really) have un-writable error
messages. This mucks with our code, since we rewrite it to signal user
errors, give context, etc.

If an error's message isn't writable, duplicate the error with one that
is.

Fixes https://github.com/ampproject/amphtml/issues/8917.

### CODE REVIEW

File: src/log.js [hunk 1]
```diff
@@ -417,6 +418,30 @@ function pushIfNonEmpty(array, val) {
...
+    return error;
+  }
+
+  const e = new Error(error.message);
+  // Copy all the extraneous things we attach.
+  for (const prop in error) {
+    e[prop] = error[prop];
+  }
+  // Ensure these are copied.
+  e.stack = error.stack;
```
Comment (dvoytenko) [inline src/log.js]: Is this really a thing?

Comment (jridgewell) [inline src/log.js]: I'm not 100% sure that `stack` is an enumerable property in all browsers, so it seemed like a good defense.

Or, are you asking about overriding the stack? In that case, yup, it's a thing. It's used a lot to clean stack traces, avoiding "internal" frames.

**Comment by muxin** [comment]
Do I need to cherry pick this to 1% canary (to be promoted to prod) and opt in canary (to be deployed)?

**Comment by jridgewell** [comment]
Nah.

**Pull request merged**